WordPress/Sniffs/Theme/NoHardCodedUrlsSniff.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * WordPress_Sniffs_Theme_NoHardCodedUrlsSniff File Doc Comment
+ *
+ * @category WordPress_Sniffs_Theme_NoHardCodedUrlsSniff
+ * @package  PHP_Codesniffer
+ * @author   khacoder
+ * @license  http://www.gnu.org/copyleft/gpl.html GNU General Public License
+ * @link     http://kevinsspace.ca
+ */
+
+/**
+ * WordPress_Sniffs_Theme_NoHardCodedUrlsSniff Class Doc Comment
+ *
+ * WARNING Verify that no hard-coded URLs are found. Manual verification necessary.
+ * An exception is made for the Author URI and (Theme) URI as set in the style.css header
+ * as well as links to wordpress.org. Links in the text of PHP/JS comments should be
+ * excluded from this check.
+ *
+ * @category WordPress_Sniffs_Theme_NoHardCodedUrlsSniff
+ * @package  PHP_Codesniffer
+ * @author   khacoder
+ * @license  http://www.gnu.org/copyleft/gpl.html GNU General Public License
+ * @link     http://kevinsspace.ca
+ */
+class WordPress_Sniffs_Theme_NoHardCodedUrlsSniff implements PHP_CodeSniffer_Sniff {
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_CONSTANT_ENCAPSED_STRING,
+			T_INLINE_HTML,
+		);
+	}//end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+		$tokens = $phpcsFile->getTokens();
+		$token  = $tokens[ $stackPtr ];
+
+		/** If the token contains this string, exclude it
+		 * ToDo - How can we account for an allowed footer link?
+		 * If we require the footer link to be an Author URI or Theme URI it's possible
+		 */
+		 $allowed_strings = array(
+				'gmpg.org',
+				'wordpress.org',
+				'wordpress.com',
+				'schema.org',
+				'fonts.googleapis',
+		 );
+		// regex borrowed from themecheck (borrowed from TAC), modified for url by Kevin.
```

Comment:
If you want to mentioned that you modified it, I would use your username as in the future we may not know which Kevin though we would have git history which we can see who contributed to the code.

FYI (TAC) is Theme Authenticity Checker